arbnode/maintenance.go

```diff
@@ -136,41 +141,102 @@ func wentPastTimeOfDay(before time.Time, after time.Time, timeOfDay int) bool {
 	return prevMinutes < dbCompactionMinutes && newMinutes >= dbCompactionMinutes
 }
 
-func (mr *MaintenanceRunner) maybeRunMaintenance(ctx context.Context) time.Duration {
+func (mr *MaintenanceRunner) GetLastMaintenanceTime() (time.Time, error) {
+	milli := mr.lastMaintenance.Load()
+	if milli == 0 {
+		return time.Time{}, errors.New("maintenance running")
+	}
+	return time.UnixMilli(milli), nil
+}
+
+func (mr *MaintenanceRunner) setMaintenanceDone() {
+	milli := time.Now().UnixMilli()
+	prev := mr.lastMaintenance.Swap(milli)
+	if prev != 0 {
```

Comment:
prev will always be equals to zero here

Reply:
nitpick: I would log an error instead of a warn then, since if this occur we have a bug